service/controller/v13/cloudconfig/master_template.go

```diff
@@ -68,7 +72,9 @@ type RandomKeyTmplSet struct {
 
 type MasterExtension struct {
 	certs            legacy.CompactTLSAssets
+	ctx              context.Context
```

Comment:
I think passing the context to the struct is a bad thing. E.g. ancelling this context can have unexpected consequences. I think context is designed to be request scope, so it should be only passed as a first argument to the function call.

Reply:
My suggestion is to pass the context to `Files` and then `templateData` method.

Reply:
The problem I see with that approach is that `Files` is called while rendering the template, for the master template here https://github.com/giantswarm/k8scloudconfig/blob/master/v_3_3_3/master_template.go#L1810. Pls correct me if I'm wrong but according to https://golang.org/pkg/text/template/#hdr-Arguments we can only use niladic methods on templates.

Reply:
You're right I missed that. IMO it's better to use context.TODO in such case.

Reply:
That's way better!